### ISSUE TITLE
Connector Builder UI: JWT Authenticator

### ISSUE THREAD

**Issue by lmossman** [airbytehq/airbyte#27903]
## Context
Blocked by https://github.com/airbytehq/airbyte/issues/25728

Add support for the JWT Authenticator to the Connector Builder UI which has the following inputs:
- Payload: JSON editor which supports interpolation (stretch goal: allow switching between list of json key-value pairs and full editor)
- Headers: Existing key-value pairs input used for the requester
- Algorithm: Dropdown
- Expiration time: Text input
- Request authentication: Like for generalized session token authenticator, oneOf for apikey/bearer, defaults to bearer

Automatically creates one inferred string input for the key

## Acceptance Criteria
- User can configure a JWT authenticator in the connector builder UI
- Can switch between manifest and UI with JWT authenticator converted between both correctly
